docs/api-usage.rst

```diff
@@ -93,20 +93,26 @@ Examples:
 .. code-block:: python
 
    # list all the projects
-   projects = gl.projects.list()
+   projects = gl.projects.list(as_list=False)
    for project in projects:
        print(project)
 
    # get the group with id == 2
    group = gl.groups.get(2)
-   for project in group.projects.list():
+   for project in group.projects.list(as_list=False):
```

Comment:
Maybe we can just have `all=True` everywhere, seems like we mix it now a bit. And just explain `as_list=False` for when they want an efficient generator (if we don't already below).

This is the intro page for the API and `list(as_list=False)` might look funny when it's there before it's explained.

Reply:
The problem with this example and `all=True` is if they run it on `gitlab.com` it will take a LONG time.

Reply:
Ah that's true, but we have the same with then endlessly printing over the iterator. Maybe it's clearer if we give an example each without, with all=True and with as_list=False explicitly, like we do with the token examples?